server/src/bigquery/BigQueryContext.ts

```diff
@@ -0,0 +1,120 @@
+import { AnalyzeResponse__Output } from '@fivetrandevelopers/zetasql/lib/types/zetasql/local_service/AnalyzeResponse';
+import { err, ok, Result } from 'neverthrow';
+import { TextDocument } from 'vscode-languageserver-textdocument';
+import { DbtProfileCreator, DbtProfileError, DbtProfileSuccess } from '../DbtProfileCreator';
+import { DestinationDefinition } from '../DestinationDefinition';
+import { SchemaTracker } from '../SchemaTracker';
+import { YamlParser } from '../YamlParser';
+import { ZetaSqlWrapper } from '../ZetaSqlWrapper';
+import { BigQueryClient } from './BigQueryClient';
+
+interface PresentBigQueryContext {
+  schemaTracker: SchemaTracker;
+  destinationDefinition: DestinationDefinition;
+  zetaSqlWrapper: ZetaSqlWrapper;
+}
+
+export class BigQueryContext {
+  private static CONTEXT_NOT_INITIALIZED_ERROR = 'Context is not initialized.';
+
+  private dbtProfileCreator: DbtProfileCreator;
+
+  private present?: boolean;
+  private presentBigQueryContext?: PresentBigQueryContext;
```

Comment:
Can we just put all fields to `BigQueryContext`? Looks like it will simplify it a bit.

Reply:
Fixed.